### PR TITLE
Make it build 1.7.10 version and small fix

### PR DESCRIPTION
**Pull request by AlexCatze**
1) Use GTNH ForgeGradle
2) Use decompiled sources from last 1.7.10 version, instead of newer one
3) Add translations support for labels

### CODE REVIEW

File: gradle.properties [hunk 1]
```diff
@@ -1,3 +1,116 @@
...
+# Define a source file of your project with:
+# public static final String VERSION = "GRADLETOKEN_VERSION";
+# The string's content will be replaced with your mod's version when compiled. You should use this to specify your mod's
+# version in @Mod([...], version = VERSION, [...])
+# Leave these properties empty to skip individual token replacements
+replaceGradleTokenInFile = Tags.java
+gradleTokenModId = GRADLETOKEN_MODID
+gradleTokenModName = GRADLETOKEN_MODNAME
+gradleTokenVersion = GRADLETOKEN_VERSION
+gradleTokenGroupName = GRADLETOKEN_GROUPNAME
```
Comment (glowredman) [inline gradle.properties]:
```
replaceGradleTokenInFile = CustomMainMenu.java
gradleTokenModId =
gradleTokenModName =
gradleTokenVersion = GRADLETOKEN_VERSION
gradleTokenGroupName =
```

File: LICENSE-template [hunk 2]
```diff
@@ -0,0 +1,21 @@
+MIT License
```
Comment (glowredman) [inline LICENSE-template]: Delete this file

File: LICENSE.txt [hunk 3]
```diff
@@ -1,9 +0,0 @@
-MIT
```
Comment (glowredman) [inline LICENSE.txt]: Keep this file.

File: settings.gradle [hunk 4]
```diff
@@ -0,0 +1,10 @@
+plugins {
+    id 'com.diffplug.blowdryerSetup' version '1.6.0'
+}
+
+apply plugin: 'com.diffplug.blowdryerSetup'
+
+blowdryerSetup {
+    //github('GTNewHorizons/ExampleMod1.7.10', 'tag', '0.1.5')
+    devLocal '.' // Use this when testing config updates locally
```
Comment (glowredman) [inline settings.gradle]: Uncomment line 8, comment line 9.

File: src/main/java/lumien/custommainmenu/CustomMainMenu.java [hunk 5]
```diff
@@ -1,100 +1,101 @@
...
-		catch (Exception e)
-		{
-			e.printStackTrace();
-
-			EVENT_HANDLER.displayMs = System.currentTimeMillis();
-			logger.log(Level.ERROR, "Error while loading new config file, trying to keep the old one loaded.");
-			config = backup;
-		}
-	}
+@Mod(modid = "CustomMainMenu", name = "Custom Main Menu", version = "1.9.2")
```
Comment (glowredman) [inline src/main/java/lumien/custommainmenu/CustomMainMenu.java]:
```suggestion
@Mod(modid = CustomMainMenu.MOD_ID, name = CustomMainMenu.MOD_NAME, version = CustomMainMenu.VERSION)
```

File: src/main/java/lumien/custommainmenu/CustomMainMenu.java [hunk 6]
```diff
@@ -1,100 +1,101 @@
...
+public class CustomMainMenu {
+    public static final String MOD_ID = "CustomMainMenu";
+    public static final String MOD_NAME = "Custom Main Menu";
+    public static final String MOD_VERSION = "1.9.2";
```
Comment (glowredman) [inline src/main/java/lumien/custommainmenu/CustomMainMenu.java]:
```suggestion
    public static final String MOD_VERSION = "GRADLETOKEN_VERSION";
```

File: README.md [hunk 7]
```diff
@@ -1,2 +0,0 @@
-Source Code for the Custom Main Menu Minecraft Mod
```
Comment (glowredman) [inline README.md]: Why did you remove this?

File: gradle.properties [hunk 8]
```diff
@@ -1,3 +1,116 @@
-# Sets default memory used for gradle commands. Can be overridden by user or command line properties.
-# This is required to provide enough memory for the Minecraft decompilation process.
-org.gradle.jvmargs=-Xmx3G
+modName = CustomMainMenu
```
Comment (glowredman) [inline gradle.properties]:
```suggestion
modName = Custom Main Menu
```

File: gradle.properties [hunk 9]
```diff
@@ -1,3 +1,116 @@
...
+gradleTokenGroupName = GRADLETOKEN_GROUPNAME
+
+# In case your mod provides an API for other mods to implement you may declare its package here. Otherwise, you can
+# leave this property empty.
+# Example value: apiPackage = api + modGroup = com.myname.mymodid -> com.myname.mymodid.api
+apiPackage =
+
+# Specify the configuration file for Forge's access transformers here. It must be placed into /src/main/resources/META-INF/
+# Example value: mymodid_at.cfg
+accessTransformersFile = CustomMainMenu_At.cfg
```
Comment (glowredman) [inline gradle.properties]: The access transformer seems not to be needed at all...

File: src/main/resources/META-INF/CustomMainMenu_At.cfg [hunk 10]
```diff
@@ -18,7 +18,6 @@ protected net.minecraft.client.gui.GuiMainMenu func_73972_b(II)V #addDemoButtons
 protected net.minecraft.client.gui.GuiMainMenu func_73969_a(II)V #addDemoButtons
 protected net.minecraft.client.gui.GuiMainMenu func_73971_c(IIF)V #renderSkybox
 
-
 #public net.minecraft.client.renderer.entity.RenderPlayer func_96449_a(Lnet/minecraft/entity/EntityLivingBase;DDDLjava/lang/String;FD)V #renderLivingLabel
 #public net.minecraft.util.ResourceLocation field_110626_a #resourceDomain
-#public net.minecraft.util.ResourceLocation field_110625_b #resourcePath
+#public net.minecraft.util.ResourceLocation field_110625_b #resourcePath
```
Comment (glowredman) [inline src/main/resources/META-INF/CustomMainMenu_At.cfg]: You can delete this file, the access transformer seems not to be needed at all...

**Review by glowredman — CHANGES_REQUESTED** [review]
Every class has this header:
```java
/*
 * Decompiled with CFR 0.148.
 */
```
sometimes even with commented out imports. Please delete these.

Another thing: you shouldn't combine two completely unrelated things in a PR (decompilation and addition of a feature), these should be done in two seperate PRs so they can be reviewed better.

File: LICENSE-template (hunk 2 above)
Comment (AlexCatze) [reply LICENSE-template]: Fixed

File: LICENSE.txt (hunk 3 above)
Comment (AlexCatze) [reply LICENSE.txt]: Fixed

File: gradle.properties (hunk 1 above)
Comment (AlexCatze) [reply gradle.properties]: Fixed

File: settings.gradle (hunk 4 above)
Comment (AlexCatze) [reply settings.gradle]: Fixed

File: README.md (hunk 7 above)
Comment (AlexCatze) [reply README.md]: I just deleted all files in project and created it from scratch. So i lost this file. Restored id.

File: src/main/resources/META-INF/CustomMainMenu_At.cfg (hunk 10 above)
Comment (AlexCatze) [reply src/main/resources/META-INF/CustomMainMenu_At.cfg]: Fixed

File: gradle.properties (hunk 9 above)
Comment (AlexCatze) [reply gradle.properties]: Fixed

File: settings.gradle.example [hunk 11]
```diff
@@ -0,0 +1,10 @@
+plugins {
```
Comment (glowredman) [inline settings.gradle.example]: Delete this file

File: src/main/java/lumien/custommainmenu/CustomMainMenu.java [hunk 12]
```diff
@@ -1,100 +1,81 @@
...
-			EVENT_HANDLER.displayMs = System.currentTimeMillis();
-			logger.log(Level.ERROR, "Error while loading new config file, trying to keep the old one loaded.");
-			config = backup;
-		}
-	}
+@Mod(modid = CustomMainMenu.MOD_ID, name = CustomMainMenu.MOD_NAME, version = CustomMainMenu.MOD_VERSION)
+public class CustomMainMenu {
+    public static final String MOD_ID = "GRADLETOKEN_MODID";
+    public static final String MOD_NAME = "GRADLETOKEN_MODNAME";
+    public static final String MOD_VERSION = "GRADLETOKEN_VERSION";
```
Comment (glowredman) [inline src/main/java/lumien/custommainmenu/CustomMainMenu.java:23]: I did not mean replace modid and name aswell...

**Review by glowredman — CHANGES_REQUESTED** [review]

File: build.gradle [hunk 13]
```diff
@@ -1,81 +1,1103 @@
+//version: 1667597057
```
Comment (glowredman) [inline build.gradle]: This is not the newest version, run the gradle task `updateBuildScript` to update

File: src/main/java/lumien/custommainmenu/CustomMainMenu.java (hunk 12 above)
Comment (AlexCatze) [reply src/main/java/lumien/custommainmenu/CustomMainMenu.java:23]: Yeah, but, why not?

File: build.gradle (hunk 13 above)
Comment (AlexCatze) [reply build.gradle]: Fixed

File: src/main/java/lumien/custommainmenu/CustomMainMenu.java (hunk 12 above)
Comment (glowredman) [reply src/main/java/lumien/custommainmenu/CustomMainMenu.java:23]: The modid should never, ever be changed because everything you register depends on it. Exposing it via a config is therefore a bit counterproductive and even if it is never touched there, it still makes no sense to always replace a token with the same value (especially because gradle occasionally fails to do this replacement)...
Modname and groupname (which you specified in gradle.properties but didn't use here...) is *about* the same (I'm simplifying here, there is more to it).

Comment (AlexCatze) [reply src/main/java/lumien/custommainmenu/CustomMainMenu.java:23]: Fixed. But isn`t it bad to have same data in two places (gradle.properties and code) instead of moving it to one constant?

**Review by glowredman — APPROVED** [review]

File: src/main/java/lumien/custommainmenu/CustomMainMenu.java (hunk 12 above)
Comment (glowredman) [reply src/main/java/lumien/custommainmenu/CustomMainMenu.java:23]: Usually yes, imo it is outweighed in this case though.
